### PR TITLE
add: local initialisation logic

### PR DESCRIPTION
**Pull request by 07souravkunda**

### CODE REVIEW

File: nightwatch/globals.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const LocalTunnel = require('../src/local-tunnel');
+
+const localTunnel = new LocalTunnel();
+
+module.exports = {
+  beforeEach(settings) {
+    console.log(settings.hello, 'hello')
```
Comment (karanshah-browserstack) [inline nightwatch/globals.js:7]: Remove console log

File: nightwatch/globals.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      process.env.BROWSERSTACK_LOCAL_ENABLED = true;
+      if (localTunnel._localOpts.localIdentifier) {
+        process.env.BROWSERSTACK_LOCAL_IDENTIFIER = localTunnel._localOpts.localIdentifier;
+        settings.desiredCapabilities['bstack:options'].localIdentifier = localTunnel._localOpts.localIdentifier;
+      }
+    }
+  },
+
+  async after() {
+    console.log('Stopping local');
```
Comment (karanshah-browserstack) [inline nightwatch/globals.js:24]: Remove console log or print meaningful message

Comment (karanshah-browserstack) [reply nightwatch/globals.js:24]: Stopping BrowserStack Local ...

File: src/local-tunnel.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        console.log('Local started successfully');
+      } catch (err) {
+        console.log('Local failed with error: ', err);
+      }
+    }
+  }
+
+  async stop() {
+    try {
+      this._localStarted && await util.promisify(this._localTunnel.stop.bind(this._localTunnel))();
```
Comment (karanshah-browserstack) [inline src/local-tunnel.js:37]: Add handling for abrupt closes

File: src/utils/helper.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  return `${formattedDate}_${hostname}_${randomChars}`;
+}
+
+exports.isUndefined = value => (value === undefined || value === null);
+
+exports.isObject = value => (!this.isUndefined(value) && value.constructor === Object);
+
+exports.getAccessKey = (settings) => {
+  let accessKey = null;
+  if (this.isObject(settings.desiredCapabilities)) {
```
Comment (karanshah-browserstack) [inline src/utils/helper.js:24]: Fetching key from env vars? Or this would be set inside the options beforehand?

**Review by karanshah-browserstack — CHANGES_REQUESTED** [review]

File: src/utils/helper.js (hunk 4 above)
Comment (07souravkunda) [reply src/utils/helper.js:24]: this would be set inside options by nightwatch

**Review by karanshah-browserstack — APPROVED** [review]
LGTM

File: nightwatch/globals.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      }
+    }
+  },
+
+  async after() {
+    localTunnel.stop();
+  },
+
+  beforeChildProcess(settings) {
+    process.env.BROWSERSTACK_LOCAL_ENABLED == "true" &&
```
Comment (swrdfish) [inline nightwatch/globals.js:25]: These lines are not readable.

File: nightwatch/globals.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+const LocalTunnel = require('../src/local-tunnel');
+
+const localTunnel = new LocalTunnel();
+
+module.exports = {
+  async before(settings) {
+    localTunnel.configure(settings);
+    await localTunnel.start();
+    if (localTunnel._localStarted) {
```
Comment (swrdfish) [inline nightwatch/globals.js:9]: why do we need `_localStarted` ? Can't we use simple try catch?

File: src/local-tunnel.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    if (helper.isUndefined(this._localOpts)) {
+      this._localOpts = {};
+    }
+
+    if (helper.isUndefined(this._localOpts.localIdentifier)) {
+      this._localOpts.localIdentifier = helper.generateLocalIdentifier();
+    }
+    this._key = helper.getAccessKey(settings);
+    if(this._settings.local || this._settings.browserstackLocal)
+      this._localTunnel = new BStackLocal();
```
Comment (swrdfish) [inline src/local-tunnel.js:20]: Missing curly braces around if statement

**Review by swrdfish — CHANGES_REQUESTED** [review]

File: nightwatch/globals.js (hunk 6 above)
Comment (07souravkunda) [reply nightwatch/globals.js:9]: Inside `start` method checking for accessKey available and keeping `_localStarted`, which is used at `stop`

**Review by karanshah-browserstack — APPROVED** [review]

File: nightwatch/globals.js (hunk 6 above)
Comment (swrdfish) [reply nightwatch/globals.js:9]: why not error out directly if `localTunnel.start` fails?

**Review by karanshah-browserstack — CHANGES_REQUESTED** [review]

**Review by karanshah-browserstack — APPROVED** [review]

**Review by swrdfish — APPROVED** [review]
LGTM

**Review by beatfactor — CHANGES_REQUESTED** [review]
please add:
- .gitignore as in the main nightwatch repo
- eslint task (use the .eslintrc file from the nightwatch repo)
- unit tests
- github actions build

**Pull request merged**